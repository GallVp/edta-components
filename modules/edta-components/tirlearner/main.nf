process TIRLEARNER {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/tir-learner:3.0.1--hdfd78af_0':
        'biocontainers/tir-learner:3.0.1--hdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)
    val species

    output:
    tuple val(meta), path("${prefix}.fa")           , emit: fasta
    tuple val(meta), path("${prefix}.gff3")         , emit: gff
    tuple val(meta), path("${prefix}.filtered.fa")  , emit: filtered_fasta
    tuple val(meta), path("${prefix}.filtered.gff3"), emit: filtered_gff
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args    = task.ext.args     ?: ''
    prefix      = task.ext.prefix   ?: "${meta.id}"
    if ( "$fasta" == "${prefix}.fa" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    TIR-Learner \\
        -f $fasta \\
        -s $species \\
        -t $task.cpus \\
        -o $prefix \\
        $args

    mv "${prefix}/TIR-Learner-Result/TIR-Learner_FinalAnn.fa"           "${prefix}.fa"
    mv "${prefix}/TIR-Learner-Result/TIR-Learner_FinalAnn.gff3"         "${prefix}.gff3"

    mv "${prefix}/TIR-Learner-Result/TIR-Learner_FinalAnn_filter.fa"    "${prefix}.filtered.fa"
    mv "${prefix}/TIR-Learner-Result/TIR-Learner_FinalAnn_filter.gff3"  "${prefix}.filtered.gff3"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        TIR-Learner: \$(TIR-Learner -v | head -n 1 | sed 's/TIR-Learner //')
    END_VERSIONS
    """

    stub:
    def args    = task.ext.args ?: ''
    prefix      = task.ext.prefix ?: "${meta.id}"
    if ( "$fasta" == "${prefix}.fa" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    touch "${prefix}.fa"
    touch "${prefix}.gff3"

    touch "${prefix}.filtered.fa"
    touch "${prefix}.filtered.gff3"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        TIR-Learner: \$(TIR-Learner -v | head -n 1 | sed 's/TIR-Learner //')
    END_VERSIONS
    """
}
