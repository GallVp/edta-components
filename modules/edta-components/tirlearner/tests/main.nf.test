nextflow_process {

    name "Test Process TIRLEARNER"
    script "../main.nf"
    process "TIRLEARNER"

    tag "modules"
    tag "modules_nfcore"
    tag "tirlearner"
    tag "nf-core/gunzip/main"

    setup {
        run('GUNZIP') {
            script "../../../nf-core/gunzip/main"

            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['actinidia_chinensis']['genome']['genome_1_fasta_gz'], checkIfExists: true)
                ]
                """
            }
        }
    }

    test("actinidia_chinensis-genome_1_fasta_gz-success") {

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = 'others'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = 'others'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
