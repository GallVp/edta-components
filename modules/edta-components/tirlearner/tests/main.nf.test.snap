{
    "actinidia_chinensis-genome_1_fasta_gz-success": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,90bf040e7b29763351237207297dc801"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,90bf040e7b29763351237207297dc801"
                    ]
                ],
                "4": [
                    "versions.yml:md5,3d4a9683fc634e22fd79a4fbe0b23a30"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                    ]
                ],
                "filtered_fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                    ]
                ],
                "filtered_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,90bf040e7b29763351237207297dc801"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,90bf040e7b29763351237207297dc801"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3d4a9683fc634e22fd79a4fbe0b23a30"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T16:35:56.356956"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,3d4a9683fc634e22fd79a4fbe0b23a30"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "filtered_fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "filtered_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3d4a9683fc634e22fd79a4fbe0b23a30"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T16:36:08.718538"
    }
}